File: 121.BestTimetoBuyandSellStock/main.cpp
#include<iostream>
#include<vector>


using std::cout, std::max, std::endl, std::vector;


class Solution {
 public:
  int maxProfit(vector<int> prices) {
    int biggest_diff = INT_MIN;
    int current_max = INT_MIN;
    for (int i = prices.size() - 1; i >= 0; --i) {
      current_max = max(current_max, prices[i]);
      biggest_diff = max(current_max - prices[i], biggest_diff);
    }
    return biggest_diff;
  }
};



int main() {
  vector<int> prices = {7, 1, 5, 3, 6, 4};
  cout << Solution().maxProfit(prices);
}

File: 121.BestTimetoBuyandSellStock/main.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minprice = 999999
        maxprofit = 0
        for price in prices:
            if price < minprice:
                minprice = price
            elif price - minprice > maxprofit:
                maxprofit = price - minprice
        return maxprofit
